src/coreComponents/fieldSpecification/FieldSpecificationManager.cpp

```diff
@@ -64,6 +64,133 @@ void FieldSpecificationManager::expandObjectCatalogs()
+    {
+      isTargetSetNonEmpty[setNames[i]] = 0;
+      isTargetSetCreated[setNames[i]] = 0;
+    }
+
+    // Step 2: apply the boundary condition
+
+    fs.apply< dataRepository::Group >( mesh,
+                                       [&]( FieldSpecificationBase const &,
+                                            string const & setName,
+                                            SortedArrayView< localIndex const > const & targetSet,
+                                            Group & targetGroup,
+                                            string const fieldName )
+    {
+      dataRepository::InputFlags const flag = fs.getWrapper< string >( FieldSpecificationBase::viewKeyStruct::fieldNameString() ).getInputFlag();
+
+      // 2.a) If we enter this loop, we know that the set has been created
+      //      Fracture/fault sets are created later and the "apply" call silently ignores them
+      isTargetSetCreated.at( setName ) = 1;
+
+      // 2.b) If the fieldName is registered on this target, we record it
+      //      Unfortunately, we need two exceptions:
+      //       - FieldSpecification that do not target a field, like Aquifer, Traction, Equilibrium, etc. For these, the check is not
+      // necessary (the user cannot mess up)
+      //       - Face boundary conditions that target cell-based quantities, like the face BC of the flow solvers
+      if( targetGroup.hasWrapper( fieldName ) ||
+          flag == InputFlags::FALSE || // no need to check input if the input flag is false (Aquifer, Traction, Equilibrium do not target a
+                                       // field)
+          targetGroup.getName() == MeshLevel::groupStructKeys::faceManagerString() ) // the field names of the face BCs are not always
+                                                                                     // registered on
+                                                                                     // the faceManager...
+      {
+        isFieldNameFound = 1;
+      }
+
+      // 2.c) If the target set is not empty, we record it
+      //      Fracture/fault sets are sometimes at initialization and the "apply" call silently ignores them
+      if( targetSet.size() > 0 )
+      {
+        isTargetSetNonEmpty.at( setName ) = 1;
+      }
+    } );
+
+    // Step 3: MPI synchronization
+
+    isFieldNameFound = MpiWrapper::max( isFieldNameFound );
+
+    bool areAllSetsEmpty = true;
+    for( std::pair< string const, localIndex > & mapEntry : isTargetSetNonEmpty )
+    {
+      mapEntry.second = MpiWrapper::max( mapEntry.second );
+      if( mapEntry.second == 1 )
+      {
+        areAllSetsEmpty = false;
+      }
+    }
+    bool areAllSetsMissing = true;
+    for( std::pair< string const, localIndex > & mapEntry : isTargetSetCreated )
+    {
+      mapEntry.second = MpiWrapper::max( mapEntry.second );
+      if( mapEntry.second == 1 )
+      {
+        areAllSetsMissing = false;
+      }
+    }
```

Comment:
Like above, maybe could you go the `mpi` "gathering" in a first loop, then analyze the result to set the value of `areAllSetsMissing` in a second loop, to better separate the two steps? It's a tad longer, but maybe it's a bit clearer, and with the `mpi` involved, one never knows. Up to you.

Would this be equivalent?
```suggestion
    for( std::pair< string const, localIndex > & mapEntry : isTargetSetCreated )
    {
      mapEntry.second = MpiWrapper::max( mapEntry.second );
    }
    bool areAllSetsMissing = true;
    for( std::pair< string const, localIndex > const & mapEntry : isTargetSetCreated )
    {
      if( mapEntry.second == 1 )
      {
        areAllSetsMissing = false;
        break;
      }
    }
```

Reply:
Done